lntest/itest/rest_api.go

```diff
@@ -0,0 +1,352 @@
+// +build rpctest
+
+package itest
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/gorilla/websocket"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/autopilotrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/verrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	insecureTransport = &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	restClient = &http.Client{
+		Transport: insecureTransport,
+	}
+	jsonMarshaler = &jsonpb.Marshaler{
+		EmitDefaults: true,
+		OrigName:     true,
+		Indent:       "    ",
+	}
+	urlEnc          = base64.URLEncoding
+	webSocketDialer = &websocket.Dialer{
+		HandshakeTimeout: 45 * time.Second,
+		TLSClientConfig:  insecureTransport.TLSClientConfig,
+	}
+	resultPattern = regexp.MustCompile("{\"result\":(.*)}")
+)
+
+// testRestApi tests that the most important features of the REST API work
```

Comment:
Nice! I wonder if we should start to run these as a distinct job on Github. Maybe somethign to revisit later if we think we need to add a significant amount of new test cases. With this as is, we at least now have some basic sanity checks to ensure we don'[t break the RESt behavior silently for our users.

Reply:
They run pretty fast as all of these calls are pretty light-weight.
These tests should greatly increase our confidence when updating the `grpc-web` library or changing other things like JSON serializer or similar.
Added a test for CORS as well and fixed a few errors in the process.